awx/main/tasks.py

```diff
@@ -979,7 +979,7 @@ def build_execution_environment_params(self, instance, private_data_dir):
 
         if instance.execution_environment.credential:
             cred = instance.execution_environment.credential
-            if cred.has_inputs(field_names=('host', 'username', 'password')):
+            if all([cred.has_input(field_name) for field_name in ('host', 'username', 'password')]):
```

Comment:
It looks like the expected behavior is to throw a runtime error when a credential is assigned to an EE and that credential doesn't have the required fields ('host', 'username', 'password'). The method we were using for that check can't return a False, though (it only raises an exception or returns True), so that exception was unreachable.